src/components/ManagingHouseForm.vue

```diff
@@ -0,0 +1,333 @@
+import useVuelidate from '@vuelidate/core';
+import Button from 'primevue/button';
+import InputText from 'primevue/inputtext';
+import { houseValidations } from '@/utils/house-validations';
+import { HousesActionsEnum, HousesGettersEnum } from '@/store/houses/types';
+import { StoreModuleEnum } from '@/store/types';
+import { AddressModel } from '@/shared/models/address.model';
+import { HouseModel } from '@/shared/models/house.model';
+import { mapGetters } from 'vuex';
+
+export default defineComponent({
+  name: 'ManagingHouseForm',
+  components: {
+    Button,
+    InputText,
+  },
+  props: {
+    cooperationId: {
+      type: Number,
+      required: true,
+    },
+    houseData: {
+      type: HouseModel,
+      required: true,
+    },
+    isEditHouse: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    isAddHouse: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      houseData: {
+        flatQuantity: null,
+        houseArea: null,
+        adjoiningArea: null,
+        address: {
+          region: '',
+          city: '',
+          district: '',
+          street: '',
+          houseBlock: '',
+          houseNumber: '',
+          zipCode: '',
+        } as AddressModel,
+      } as HouseModel,
+
+      isLoaded: false,
+      v$: useVuelidate(),
+    };
+  },
+  validations() {
+    return {
+      houseData: houseValidations,
+    };
+  },
+  async mounted() {
+    if (this.$props.isEditHouse) {
+      const payload = {
+        cooperationId: this.$props.cooperationId,
+        houseId: this.$props.houseData.id,
+      };
+
+      await this.$store.dispatch(`${StoreModuleEnum.housesStore}/${HousesActionsEnum.GET_HOUSE_BY_ID}`, payload);
+      this.initData();
+    }
+    this.isLoaded = true;
+  },
+  methods: {
+    initData() {
+      this.houseData.flatQuantity = this.selectedHouseInfo?.flatQuantity;
+      this.houseData.houseArea = this.selectedHouseInfo?.houseArea;
+      this.houseData.adjoiningArea = this.selectedHouseInfo?.adjoiningArea;
+      this.houseData.address = JSON.parse(JSON.stringify(this.selectedHouseInfo?.address ?? ({} as AddressModel)));
+    },
+    async manageHouseInfo() {
+      if (this.$props.isEditHouse) {
+        const payload = {
+          ids: {
+            cooperationId: this.$props.cooperationId,
+            houseId: this.$props.houseData.id,
+          },
+          houseData: {
+            id: this.$props.houseData.id,
+            ...this.houseData,
+          },
+        };
+
+        await this.$store.dispatch(`${StoreModuleEnum.housesStore}/${HousesActionsEnum.EDIT_HOUSE}`, payload);
+        this.showSuccessOperation(`Дані про будинок з ID ${this.$props.houseData.id!} змінено!`);
+      } else if (this.$props.isAddHouse) {
+        const payload = {
+          cooperationId: this.$props.cooperationId,
+          houseData: { ...this.houseData },
```

Comment:
$props.houseData has all information about the house (not just 'id') doesn't it? So you don't need use getter 'getHouseInfo'? You can use information from the props.

Reply:
Yeah, as far as I got it, there are two ways for getting info about selected house, from props or getter. We can use both, but apparently, the decision with props will be better since we don't need to make one else request to server.

Reply:
It's not a final version of this task) just attempt